### PR TITLE
HEEDLS-274 Initial Menu page customisation front end

### PR DESCRIPTION
**Pull request by benjimarshall**
## Changes
Add customisation settings to initial menu page, these control whether to display course duration, and percentage
completion, and come from a JSON field in the application table.

## Testing
Add view model tests for these new settings.

## Screenshots
NB my local DB was modified to make these tests, there are no active courses with either of these flags sets in the original database.

### A course with section percentages removed
percentages_removed

### A course with overall duration removed
durations_removed

### A course with overall duration and section percentages removed
both_removed

### CODE REVIEW

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs [hunk 1]
```diff
@@ -172,6 +172,40 @@ public void Initial_menu_should_have_centre_name()
...
+
+            // When
+            var initialMenuViewModel = new InitialMenuViewModel(expectedCourseContent);
+
+            // Then
+            initialMenuViewModel.ShowTime.Should().BeTrue();
+        }
+
+        [Test]
+        public void Initial_menu_should_have_false_show_time()
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs:193]: What about having a test for false show time if the time is null? 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs]: Yes we definitely need one. Added in b172872 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs [hunk 2]
```diff
@@ -172,6 +172,40 @@ public void Initial_menu_should_have_centre_name()
...
         }
 
+        [Test]
+        public void Initial_menu_should_have_true_show_time()
+        {
+            // Given
+            const string courseSettings = "{\"lm.st\":true}"; // ShowTime = true
+            var expectedCourseContent = CourseContentHelper.CreateDefaultCourseContent(
+                courseSettings: courseSettings
+
```
[the remark this answers is not in the document]
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs]: Yep, sorry! Fixed in b172872 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs [hunk 3]
```diff
@@ -172,6 +172,40 @@ public void Initial_menu_should_have_centre_name()
...
+        }
+
+        [Test]
+        public void Initial_menu_should_have_false_show_time()
+        {
+            // Given
+            const string courseSettings = "{\"lm.st\":false}"; // ShowTime = false
+            var expectedCourseContent = CourseContentHelper.CreateDefaultCourseContent(
+                courseSettings: courseSettings
+
```
[the remark this answers is not in the document]
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/IntitialMenuViewModelTests.cs:202]: Yep, sorry! Fixed in b172872 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**